### PR TITLE
Added table support

### PR DESCRIPTION
**Pull request by dumitrubogdanmihai**
Tutorial:
- copy the _html_cals_table.css_ from XHTML framework
- apply the CSS to all documents by declaring it in Document Type Association dialog, on Author tab, under CSS section
- copy the _xhtml.jar_ from XHTML framework to have all the actions that manipulate the table (insert row, split, join, etc.)
- declare the jar in Document Type Association, under Classpath tab
- added _translation.xml_ that into i18n directory. This file contain messages presented in UI in all supported languages
- declare the _translation.xml_ file in Document Type Association, under Classpath tab
- added icons for toolbar actions in _images_ directory
- define actions to be added on toolbar in Document Type Association dialog, on Author tab, under Actions section
- add previously defined actions to toolbar from Document Type Association dialog, on Author tab, under Toolbar section
- specially for Web Author added in _web_ the _html-table-support.js_ file.
This will wrap the actions defined in Document Type Association to make context aware and to present a dialog to ask for Caption when the Insert Table action is invoked.

### CODE REVIEW

File: frameworks/legaldocml/legaldocml.framework [hunk 1]
```diff
@@ -23,8 +23,10 @@
 					</field>
 					<field name="classpath">
 						<String-array>
-							<String>${framework}/target/classes/</String>
+							<String>C:\Users\intern\Desktop\framework-extensions\legaldocml\xhtml.jar</String>
```
Comment (ctalau) [inline frameworks/legaldocml/legaldocml.framework]: It looks like a hardcoded path. I think it is not even required.

File: frameworks/legaldocml/web/html-table-support.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
+
+sync.table.HtmlExtension = sync.table.HtmlExtension || {};
+
+
+/**
+ * Constructor for the Extension.
+ *
+ * @constructor
+ */
+sync.table.HtmlExtension.Extension = function(){
```
Comment (ctalau) [inline frameworks/legaldocml/web/html-table-support.js]: Should probably be named AkomaNtosoExtension without any namespace.

File: frameworks/legaldocml/web/html-table-support.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+ * @param {sync.Editor} editor The currently created editor.
+ */
+sync.table.HtmlExtension.Extension.prototype.editorCreated = function(editor) {
+  goog.events.listen(editor, sync.api.Editor.EventTypes.ACTIONS_LOADED, function(e) {
+	  
+	
+    var actionsManager = editor.getActionsManager();
+
+    // Wrap the Insert Web Link action.
+    var insertWebLinkActionId = 'a.href';
```
Comment (ctalau) [inline frameworks/legaldocml/web/html-table-support.js]: Do we want to add the "insert link" action as well?

File: frameworks/legaldocml/web/html-table-support.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+	}
+	
+  }
+
+}
+
+// Publish the extension.
+sync.ext.Registry.extension = new sync.table.HtmlExtension.Extension();
+
+sync.actions.AkomanNtosoInsertTable = function(a, b, c, d, e, f, g) {
```
Comment (ctalau) [inline frameworks/legaldocml/web/html-table-support.js]: Fix the names of the parameters.

File: frameworks/legaldocml/web/html-table-support.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+	 sync.actions.InsertTable.call(this, a, b, c, d, e, f, g);
+	 this.editor = c;
+}
+goog.inherits(sync.actions.AkomanNtosoInsertTable, sync.actions.InsertTable);
+
+sync.actions.AkomanNtosoInsertTable.prototype.invokeOperation = function(a, b, c) {
```
Comment (ctalau) [inline frameworks/legaldocml/web/html-table-support.js]: Fix the name of the parameters.

File: frameworks/legaldocml/web/html-table-support.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+// Publish the extension.
+sync.ext.Registry.extension = new sync.table.HtmlExtension.Extension();
+
+sync.actions.AkomanNtosoInsertTable = function(a, b, c, d, e, f, g) {
+	 sync.actions.InsertTable.call(this, a, b, c, d, e, f, g);
+	 this.editor = c;
+}
+goog.inherits(sync.actions.AkomanNtosoInsertTable, sync.actions.InsertTable);
```
Comment (ctalau) [inline frameworks/legaldocml/web/html-table-support.js:138]: Do we really need to extend the default operation? I think just calling InsertFragmentOpeation would be enough.

**Review by ctalau — CHANGES_REQUESTED** [review]

File: frameworks/legaldocml/web/html-table-support.js (hunk 6 above)
Comment (dumitrubogdanmihai) [reply frameworks/legaldocml/web/html-table-support.js:138]: Only because _sync.actions.InsertTable_ show a dialog that asks for table's title.

Comment (ctalau) [reply frameworks/legaldocml/web/html-table-support.js:138]: Baga-l mereu cu titlu si terminam povestea. Asta vreau sa fac si pe DITA.

**Review by ctalau — CHANGES_REQUESTED** [review]

**Comment by dumitrubogdanmihai** [comment]
I closed this PR because there is an easier and better way to add table-related features: [web-author-CALS-table-plugin](https://github.com/oxygenxml/web-author-sample-plugins/tree/master/web-author-CALS-table-plugin). One advantage of the plugin is that it can be easily updated, compared to a bunch of files added to a framework in various places.

**Pull request closed without merging**